blocks/api/registration.js

```diff
@@ -252,7 +230,7 @@ export function getBlockType( name ) {
  * @return {Array} Block settings.
  */
 export function getBlockTypes() {
-	return Object.values( blocks );
+	return select( 'core/blocks' ).getBlockTypes();
```

Comment:
getBlockTypes is used for example to get transforms. So it seems we are saving function references in the state. Maybe we can just save the serializable block information in the state (title icon etc...), and have an external object that contains maps the block to the non-serializable information like functions in the transforms.

Reply:
Maybe we can make use of resolvers or something similar, so it looks like everything is in the store, and we can select and dispatch as we do now.

Reply:
I personally think it's ok to save non-serializable data in the state. The benefits outgrow the downsides. Having serializable information in state is only needed if you want to persist state or dispatch things over the network. I think the blocks store is not something we'd want to persist or dispatch over the network if we implement collaborative editing for instance.

Reply:
Yes, I agree saving the function references in state simplifies the things. Most sources discourage saving non-serializable info state, but objectively it looks like the only thing we lose is the ability to persist and rehydrate data https://github.com/markerikson/redux/blob/create-faq-page/docs/FAQ.md#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state, so I'm ok with doing this tradeoff as I did not found other concrete disavantages.